### PR TITLE
Switch Gimlet revision from i64 to u32

### LINKED ISSUE #6095: Gimlet baseboard revision should be unsigned
We're currently storing a Gimlet's baseboard revision as an `i64` here:

https://github.com/oxidecomputer/omicron/blob/f6d9e3c389ce228306680c56419e30506e643785/sled-hardware/types/src/lib.rs#L25

That appears to be because we read it from a devinfo property, and that library only understands signed values, for example in `di_prop_int64(3DEVINFO)`. However, both the host OS and the SP believe this to be a `u32`. Here's the host-side:

https://github.com/oxidecomputer/illumos-gate/blob/e517f5ad5a166aedb4450aee69f8b8e80cd5a8c1/usr/src/uts/oxide/sys/ipcc.h#L68-L72

And the SP side:

https://github.com/oxidecomputer/hubris/blob/fca59bdbd6e83d2fc31c7f1a14b28de463e2095b/lib/oxide-barcode/src/lib.rs#L40-L43

We should probably convert this to a `u32` in the sled agent to align with the source of truth.

### PR DESCRIPTION
**Pull request by bnaecker**
Fixes #6095

### CODE REVIEW

File: sled-storage/src/dataset.rs [hunk 1]
```diff
@@ -311,6 +311,7 @@ pub(crate) async fn ensure_zpool_has_datasets(
             if let Ok(epoch) = epoch_str.parse::<u64>() {
                 epoch
             } else {
+                println!(">>>> {epoch_str}");
```
Comment (smklein) [inline sled-storage/src/dataset.rs:314]: stray println?

**Review by smklein — APPROVED** [review]

File: sled-storage/src/dataset.rs (hunk 1 above)
Comment (bnaecker) [reply sled-storage/src/dataset.rs:314]: Yeah, nice catch thank you.

Comment (bnaecker) [reply sled-storage/src/dataset.rs]: Removed in b5c12588f

**Comment by bnaecker** [comment]
This got caught in the middle of the move to the trait-based API server for wicketd, which is substantial. I'm going to fix up locally and push another commit.

**Pull request merged**